test/valid-data/annotation-example/schema.json

```diff
@@ -7,11 +7,9 @@
     },
     "MyObject": {
       "additionalProperties": false,
-      "examples": [
-        {
-          "nested": "hello"
-        }
-      ],
+      "example": {
+        "nested": "hello"
+      },
```

Comment:
Is this actually better? The old format is more predictable and easier to process since it's consistent. 

Can you send a link to json schema spec about this tag?

Reply:
Examples does not pass validation under openapi spec, while `example` does

Reply:
Hmm, then I would say this is an issue to raise with openapi and swagger. I don't really want to diverge from JSON schema. 

I was somewhat reluctant to add the discriminatorType flag and would not like to support too many flags.

Reply:
Makes sense.

 >  would not like to support too many flags.

the open api is already there


You might close this PR if you wish, I don't think there is anymore to be done here.

Reply:
How about we add one flag for openapi support and use that for all openapi related features?